packages/cli/src/CredentialsHelper.ts

```diff
@@ -450,28 +452,34 @@ export class CredentialsHelper extends ICredentialsHelper {
 			return false;
 		}
 
-		const credential = await this.sharedCredentialsRepository.findOne({
-			where: {
-				role: 'credential:owner',
-				project: {
-					projectRelations: {
-						role: 'project:personalOwner',
-						user: {
-							role: In(['global:owner', 'global:admin']),
+		return (
+			(await this.cacheService.get(`credential-can-use-secrets:${nodeCredential.id}`, {
```

Comment:
Question: I think the cache service in memory will store the boolean values, but if using queue mode (or Redis as cache, since this can be forced via settings) it would get serialized. In this case, I think we'd end up getting a string rather than a boolean value, possibly breaking. Have you tested this scenario?

Reply:
Just tested it and it works.